### PR TITLE
Support java.util.Properties in Java 8

### PR DESCRIPTION
**Pull request by abstratt**
Hashtable's readObject() uses unsupported ObjectInputStream#readFields(), and so did Properties up to Java 8.

Issue: #24765

### CODE REVIEW

**Comment by abstratt** [comment]
@bot-gradle help

**Comment by abstratt** [comment]
@bot-gradle test QF

**Comment by abstratt** [comment]
@bot-gradle test ACC

**Review by mlopatkin — APPROVED** [review]
:100: LGTM!

File: subprojects/configuration-cache/src/test/kotlin/org/gradle/configurationcache/serialization/codecs/BaseTypeCodecTest.kt [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+class BaseTypeCodecTest : AbstractUserTypeCodecTest() {
+
+    @Test
+    fun `can handle Properties`() {
+        val properties = Properties()
+        properties.setProperty("prop1", "value1")
+        properties.setProperty("prop2", "value2")
+        configurationCacheRoundtripOf(properties).run {
+            MatcherAssert.assertThat(properties, CoreMatchers.equalTo(this))
```
Comment (bamboo) [inline subprojects/configuration-cache/src/test/kotlin/org/gradle/configurationcache/serialization/codecs/BaseTypeCodecTest.kt:34]: 💄 Statically import `assertThat` and `equalTo` for a better prose.

**Review by bamboo — CHANGES_REQUESTED** [review]

File: subprojects/configuration-cache/src/test/kotlin/org/gradle/configurationcache/serialization/codecs/BaseTypeCodecTest.kt (hunk 1 above)
Comment (abstratt) [reply subprojects/configuration-cache/src/test/kotlin/org/gradle/configurationcache/serialization/codecs/BaseTypeCodecTest.kt]: ✅

**Comment by abstratt** [comment]
@bot-gradle test and merge

**Comment by abstratt** [comment]
@bot-gradle test and merge

**Pull request merged**